File: Back-Ends/JavaScript/src/Module/Transaction/Model/transaction.model.js
const newDate = new Date().toISOString();

export default (sequelize, DataTypes) => {
  const Transactions = sequelize.define("transactions", {
    userId: {
      type: DataTypes.STRING,
      allowNull: true,
      references: {
        model: "Users",
        key: "id",
      },
      onUpdate: "CASCADE",
      onDelete: "CASCADE",
    },
    accountId: {
      type: DataTypes.STRING,
      allowNull: true,
      references: {
        model: "Accounts",
        key: "id",
      },
      onUpdate: "CASCADE",
      onDelete: "CASCADE",
    },
    amount: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    referenceNo: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    category: {
      type: DataTypes.ENUM("wallet-deposit", "group-deposit"),
      allowNull: false,
    },
    createdAt: {
      type: newDate,
      allowNull: false,
    },
    updatedAt: {
      type: newDate,
      allowNull: false,
    },
  });
  return Transactions;
};

File: Back-Ends/Golang/api/models/transaction.model.go
package models

import (
	"time"
)

type Transaction struct {
	ID          uint32    `gorm:"primary_key;auto_increment" json:"id"`
	UserId      uint32    `sql:"type:int REFERENCES users(id)" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	AccountId   uint32    `sql:"type:int REFERENCES accounts(id)" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
	Amount      string    `gorm:"size:100;not null;default:0" json:"amount"`
	ReferenceNo string    `gorm:"size:100;not null;default:0" json:"referenceNo"`
	Category    string    `gorm:"size:100;not null;default:0" json:"category"`
	CreatedAt   time.Time `gorm:"default:CURRENT_TIMESTAMP" json:"createdAt"`
	UpdatedAt   time.Time `gorm:"default:CURRENT_TIMESTAMP" json:"updatedAt"`
}
